src/Core/Managed/Shared/DataContracts/AvailabilityTelemetry.cs

```diff
@@ -0,0 +1,170 @@
+﻿namespace Microsoft.ApplicationInsights.DataContracts
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.External;
+    using System.Globalization;
+
+    /// <summary>
+    /// Telemetry type used for availability test results.
+    /// Contains a time and message and optionally some additional metadata.
+    /// </summary>
+    public sealed class AvailabilityTelemetry : ITelemetry, ISupportProperties //TODO: Should this implement IOperation like RequestTelemetry?
+    {
+        internal const string TelemetryName = "Availability";
+
+        internal readonly string BaseType = typeof(AvailabilityData).Name;
+        internal readonly AvailabilityData Data;
+        private readonly TelemetryContext context;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvailabilityTelemetry"/> class with empty properties.
+        /// </summary>
+        public AvailabilityTelemetry()
+        {
+            this.Data = new AvailabilityData();
+            this.context = new TelemetryContext(this.Data.properties);
+            this.Data.testRunId = Guid.NewGuid().ToString();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvailabilityTelemetry"/> class with empty properties.
+        /// </summary>
+        public AvailabilityTelemetry(string testName, DateTimeOffset testTimeStamp, TimeSpan duration, string runLocation, bool success, string message = null)
+        {
+            this.Data = new AvailabilityData();
+            this.context = new TelemetryContext(this.Data.properties);            
+            this.Data.testTimeStamp = testTimeStamp.ToString("o", CultureInfo.InvariantCulture);
```

Comment:
There are scenarios where someone may be computing availability with an external system (Gomez/Keynote/etc) and write a “connector” to send them to AI.  In this case the difference between when the telemetry item is submitted and when availability was computed could be very different.  The UI should be showing when the availability was measured, not when it was sent to AI.  The REST endpoint for AI currently requires a testDateTime as well.